src/Module/RemoveBookingFromCartHandler.php

```diff
@@ -0,0 +1,200 @@
+     * @since [*next-version*]
+     *
+     * @param EDD_Cart                                      $eddCart          The EDD cart instance.
+     * @param SelectCapableInterface                        $bookingsSelectRm The bookings SELECT resource model.
+     * @param DeleteCapableInterface                        $bookingsDeleteRm The bookings DELETE resource model.
+     * @param object                                        $exprBuilder      The expression builder.
+     * @param array|stdClass|ArrayAccess|ContainerInterface $cartItemConfig   The cart item data config.
+     */
+    public function __construct(
+        EDD_Cart $eddCart,
+        SelectCapableInterface $bookingsSelectRm,
+        DeleteCapableInterface $bookingsDeleteRm,
+        $exprBuilder,
+        $cartItemConfig
+    ) {
+        $this->eddCart          = $eddCart;
+        $this->bookingsSelectRm = $bookingsSelectRm;
+        $this->bookingsDeleteRm = $bookingsDeleteRm;
+        $this->exprBuilder      = $exprBuilder;
+        $this->cartItemConfig   = $cartItemConfig;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        // Get the cart item data keys from the cart item config
+        $dataKey      = $this->_containerGetPath($this->cartItemConfig, ['data', 'key']);
+        $eddBkKey     = $this->_containerGetPath($this->cartItemConfig, ['data', 'eddbk_key']);
+        $bookingIdKey = $this->_containerGetPath($this->cartItemConfig, ['data', 'booking_id_key']);
+
+        try {
+            // Get the cart index from the event (provided by the EDD hook) and normalize it
+            $cartIndex = $event->getParam(0);
+            $cartIndex = $this->_normalizeInt($cartIndex);
+
+            // Get all the cart items - EDD does not provide a single cart item getter!
+            $cartItems = $this->eddCart->get_contents();
+            // Get the booking ID from the cart item at the index
+            $bookingId = $this->_containerGetPath($cartItems, [$cartIndex, $dataKey, $eddBkKey, $bookingIdKey]);
+
+            // Build the condition for selecting the booking with the booking ID
+            $b = $this->exprBuilder;
+            $c = $b->eq($b->var('id'), $b->lit($bookingId));
+
+            // Get the bookings that match - only 1 should match
+            $bookings = $this->bookingsSelectRm->select($c);
+            if ($this->_countIterable($bookings) !== 1) {
+                return;
+            }
+
+            // If the booking has a cart status, delete it
+            $booking = reset($bookings);
+            if ($booking->get('status') === Status::STATUS_IN_CART) {
+                $this->bookingsDeleteRm->delete($c);
```

Comment:
I can see that, after jumping through hoops with EDD, you finally have the booking ID. Then you are actually removing the booking from the database after ascertaining that it exist and is in fact "in cart". So, why not abstract the database operations into a pehaps `_removeBookingFromCart()`, similarly to how you do it in the add-to-cart handler?